src/resolve/resolved-manager.c

```diff
@@ -1224,6 +1224,19 @@ DnsScope* manager_find_scope(Manager *m, DnsPacket *p) {
         return NULL;
 }
 
+DnsStream* manager_find_stream(Manager *m, DnsServer *s) {
+        DnsStream *i;
+
+        assert(m);
+        assert(s);
+
+        LIST_FOREACH(streams, i, m->dns_streams)
+                if (i->server == s)
+                        return i;
+
+        return NULL;
```

Comment:
i think it would make a ton of sense to embedd a ref to a DnsStream in the DnsServer structure (or even a `Set` object with multiple of them). Since we want to share connections now, it only makes sense if we maintain a per-server stream if we can as part of the server structure. I figure in the long run it might make sense to keep more than one stream per server open, hence maybe make this a Set* and allocate multiple streams per service.

I figure a simple algorithm of "when we want to send a new DnsPacket, look for suitable existing DnsStream objects to the DnsServer, and use the first one with an empty queue. If none has an empty queue allocate a new DnsStream object until a max of 5 streams per server is reached, in which case enqueue into the stream with the shortest queue" or so might be simple enough to implement for now and good for the beginning

If the ref to the DnsStream is embedded in the DnsServer we don't have to search the full list of streams here...